doc/userguide/configuration/suricata-yaml.rst

```diff
@@ -2035,6 +2048,83 @@ To be able to run DPDK on Intel cards, it is required to change the default Inte
+
+As mentioned previously, secondary processes must run alongside a primary process. The primary process must allocate resources. 
+These are later shared with the secondary processes. Therefore, when running Suricata as a secondary process, it needs to cooperate with a primary process.
+Main responsibilities of the primary process placed in front of Suricata should include:
+
+- allocation of memory resources
+- configuration of NICs
+- creation of communication rings
+- transfering packets from the NIC to the Suricata rings
+- transmitting packets in IPS mode
+
+Secondary applications typically do not operate with ports directly but instead, attempt to attach to DPDK rings that should be created by the primary application. 
+Suricata support arbitrary name of the receiving / transmitting rings. 
+The name of the rings specifies the `- interface:` list entry or the `copy-iface` entry. 
+As each Suricata worker reads incoming packets from a unique ring, the ring name contains static and dynamic part.
+Dynamic part, designated as `$QQQ` must be included in the ring name format. The remaining part of the ring name format is static.
+On Suricata startup, workers use the ring name format to lookup DPDK rings. 
+During this process, the dynamic part is substituted with worker's corresponding queue number (starting at 0, going upwards up to the count of Suricata worker threads). 
+
+Suricata in secondary mode does not take in respect configuration entries related to NIC setup (e.g. `mempool-size`, `mempool-cache`). 
+Instead, it relies on the primary application to do the initialization routines.
+
+Suricata can operate both in IDS and IPS mode. At the same time, Suricata can be configured in various ways to accomplish different use-cases.
+Figure :numref:`dpdk-secondary-ids` depicts a basic IDS use-case. 
+Primary application placed in front of Suricata receives packets and sends them to Suricata via DPDK rings.
+Each worker has one individual ring. It is expected from the primary application that both directions of the flow are sent to the same worker.
+After Suricata workers process the received packets, they return them to the original packet memory pool.
+
+.. _dpdk-secondary-ids:
+.. figure:: suricata-yaml/dpdk-secondary-ids.png
+    :align: center
+    :alt: Suricata as a secondary process in IDS mode
+    :figclass: align-center
+
+    `Suricata as a secondary process in IDS mode`
+
+As mentioned, Suricata supports IPS mode even when running as a secondary process. 
+The basic concept of how Suricata attaches to the rings is depicted at Figure :numref:`dpdk-secondary-ips-full`.
+Primary application creates a pair of rings for each NIC it works with.
+One ring is used for sending packets to Suricata and the other can be used to retrieve and transmit analyzed packets from Suricata through the appropriate NIC.
+However, in a classic IPS mode, there should be a pair of NICs. Entry `copy-iface` must hold ring name format of the other NIC.
```

Comment:
Can you reword this a bit? I think the second sentence is hard to understand, as is